gulpfile.js

```diff
@@ -72,21 +80,83 @@ gulp.task(
               .pipe(eslint.format())
               .pipe(eslint.failAfterError()));
 
-gulp.task(
-    'depcheck',
-    () => depcheck(__dirname, {
-            // "@types/*" dependencies are type declarations that are
-            // automatically
-            // loaded by TypeScript during build. depcheck can't detect this
-            // so we ignore them here.
-            ignoreMatches: ['@types/*', 'vinyl']
-          }).then((result) => {
-      let invalidFiles = Object.keys(result.invalidFiles) || [];
-      let invalidJsFiles = invalidFiles.filter((f) => f.endsWith('.js'));
-      if (invalidJsFiles.length > 0) {
-        throw new Error(`Invalid files: ${invalidJsFiles}`);
-      }
-      if (result.dependencies.length) {
-        throw new Error(`Unused dependencies: ${result.dependencies}`);
-      }
-    }));
+gulp.task('depcheck', () => {
+  return depcheck(__dirname, {
+    // "@types/*" dependencies are type declarations that are automatically
+    // loaded by TypeScript during build. depcheck can't detect this
+    // so we ignore them here. Same with babel plugins.
+    ignoreMatches: [
+      '@types/*', 'vinyl', 'babel-plugin-external-helpers']
+  }).then((result) => {
+    let invalidFiles = Object.keys(result.invalidFiles) || [];
+    let invalidJsFiles = invalidFiles.filter((f) => f.endsWith('.js'));
+    if (invalidJsFiles.length > 0) {
+      throw new Error(`Invalid files: ${invalidJsFiles}`);
+    }
+    if (result.dependencies.length) {
+      throw new Error(`Unused dependencies: ${result.dependencies}`);
+    }
+  });
+});
+
+/*
+ * There doesn't seem to be documentation on what helpers are available, or
+ * which helpers are required for which transforms. The
+ * source is here: https://github.com/babel/babel/blob/6.x/packages/babel-helpers/src/helpers.js
+ *
+ * This list is an educated guess at the helpers needed for our transform set
+ * of ES2015 - modules. When we switch to loose mode we should update the list.
+ *
+ * All helpers are listed here, with some commented out, so it's clear what we've excluded.
+ */
+const babelHelperWhitelist = [
+  'typeof', // Symbol support, for IE11
+  // 'jsx', // we don't support JSX
+  // 'asyncIterator', // async-iterators are not in ES2015
+  // 'asyncGenerator', // async-iterators are not in ES2015
+  // 'asyncGeneratorDelegate', // async-iterators are not in ES2015
+  // 'asyncToGenerator', // async functions are not in ES2015
+  'classCallCheck',
+  'createClass',
+  'defineEnumerableProperties',
+  'defaults', // TODO (justinfagnani): what's this for?
```

Comment:
can you respond to all these TODOs? Most of them are just looking for a confirmation/answer, we should have that before merging.